docs(README): Describe UI5 linter internals (#343)
SAP/ui5-linter: README.md

It checks JavaScript, TypeScript, XML, JSON, and other files in your project and reports findings.

## Features

> UI5 linter scans your UI5 project and detects problems that might cause it to not run smoothly with [UI5 2.x](https://community.sap.com/t5/open-source-blogs/introducing-openui5-2-x/ba-p/13580633). 

Reviewer: ```suggestion
UI5 linter scans your UI5 project and detects issues that might interfere with its smooth running with [UI5 2.x](https://community.sap.com/t5/open-source-blogs/introducing-openui5-2-x/ba-p/13580633). 
```

In this way, you can control which files the UI5 linter should process and which it should ignore.

## Internals

> UI5 linter makes use of the [TypeScript compiler](https://github.com/microsoft/TypeScript/) to parse and analyze the source code (both JavaScript as well as TypesScript) of a UI5 project. This allows a decent level of accuracy and performance.

Reviewer: ```suggestion
UI5 linter makes use of the [TypeScript compiler](https://github.com/microsoft/TypeScript/) to parse and analyze the source code (both JavaScript and TypesScript) of a UI5 project. This allows for a decent level of accuracy and performance.
```

> For this UI5 linter provides the TypeScript compiler with the SAPUI5 [type definitions](https://sap.github.io/ui5-typescript/), the compiler can then infer the usage of UI5 classes and modules in the code. UI5 linter only needs to ask the compiler whether a given function or property is deprecated and the compiler will provide it with the full deprecation text.

Reviewer: ```suggestion
For this purpose, UI5 linter provides the TypeScript compiler with the SAPUI5 [type definitions](https://sap.github.io/ui5-typescript/). The compiler can then infer the usage of UI5 classes and modules in the code. UI5 linter only needs to ask the compiler whether a given function or property is deprecated, and the compiler will provide this information with the full deprecation text.
```

> In some cases UI5 linter will transpile some of the source files before analyzing them. For example the `sap.ui.require` syntax is transpiled to `import` statements (basically ESM) so that the TypeScript compiler can understand them.

Reviewer: ```suggestion
In some cases, UI5 linter will transpile source files before analyzing them. For example, the `sap.ui.require` syntax is transpiled to `import` statements (basically ESM), so that the TypeScript compiler can understand them.
```

Similarly, UI5 XML views are transpiled to a JavaScript representation that can be analyzed in the same way as regular JavaScript code. In both cases, source maps are used to map any findings back to the correct line and column position in the original source code.

> There are additional checks build on top of the compiler API. For example the use of global variables provided by the UI5 framework (e.g. `sap.m.Button`), or for the correct usage of asynchronous component initialization.

Reviewer 2: maybe add something like 
...or checks in general that operator on non-JS/TS code (manifest.json, HTML)

Author: Done

> There are additional checks build on top of the compiler API. For example the use of global variables provided by the UI5 framework (e.g. `sap.m.Button`), or for the correct usage of asynchronous component initialization.

Reviewer: ```suggestion
There are additional checks built on top of the compiler API. This applies, for example, to the usage of global variables provided by the UI5 framework (e.g. `sap.m.Button`), or to the correct usage of asynchronous component initialization.
```

> For some checks however, the TypeScript compiler approach is sadly not sufficient. In those cases, an extract of the data structure that powers the UI5 SDK is used (the so called "api.json").

Reviewer: ```suggestion
For some checks, however, the TypeScript compiler approach is insufficient. In such cases, an extract of the data structure that powers the UI5 SDK is used (the so-called "api.json").
```
